model/common/src/icon4py/model/common/utils/gt4py_field_allocation.py

```diff
@@ -20,12 +56,15 @@ def allocate_zero_field(
     is_halfdim=False,
     dtype=ta.wpfloat,
     backend: Optional[backend.Backend] = None,
-):
-    shapex = tuple(map(lambda x: grid.size[x], dims))
-    if is_halfdim:
-        assert len(shapex) == 2
-        shapex = (shapex[0], shapex[1] + 1)
-    return gtx.as_field(dims, xp.zeros(shapex, dtype=dtype), allocator=backend)
+) -> gtx.Field:
+    def size(dim: gtx.Dimension, is_half_dim: bool) -> int:
+        if dim == dimension.KDim and is_half_dim:
+            return grid.size[dim] + 1
+        else:
+            return grid.size[dim]
```

Comment:
```suggestion
        return grid.size[dim]
```

Reply:
do we need this function inside `allocate_zero_field` or can we move it out?

Reply:
not necessarily. I thought there is no other usage for it elsewhere because the `is_half_dim` argument is pretty special to this case. But that might be wrong. It should be removed anyway once the `KDim` vs `KHalfDim` issue is resolved. 

Currently there are several solutions of the issue accross the code base.

Reply:
I extracted it and used it in the other allocator as well.